atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1629,13 +1635,13 @@ private void checkConstraints() {
 
     private void commitWrites(TransactionService transactionService) {
         if (!hasWrites()) {
-            if (hasReads()) {
+            if (hasReads() || hasAnyInvolvedTables()) {
```

Comment:
Actually this should take into account if they're serializable or not, gimme a sec...

Reply:
Fixed